spotinst/resource_spotinst_ocean_spark.go

```diff
@@ -195,27 +198,81 @@ func resourceSpotinstSparkClusterDelete(ctx context.Context, resourceData *schem
 		return diag.FromErr(err)
 	}
 
-	log.Printf("===> Cluster deleted successfully: %s <===", resourceData.Id())
+	log.Printf("===> cluster deleted successfully: %s <===", resourceData.Id())
 	resourceData.SetId("")
 
 	return nil
 }
 
 func deleteSparkCluster(ctx context.Context, resourceData *schema.ResourceData, meta interface{}) error {
 	clusterID := resourceData.Id()
+	forceDelete := false
+	if os.Getenv("TF_ACC") == "1" {
+		log.Printf("===> force delete set to true for integration tests")
+		forceDelete = true
+	}
+
 	input := &spark.DeleteClusterInput{
-		ClusterID: spotinst.String(clusterID),
+		ClusterID:   spotinst.String(clusterID),
+		ForceDelete: spotinst.Bool(forceDelete),
 	}
 
 	if json, err := commons.ToJson(input); err != nil {
 		return err
 	} else {
-		log.Printf("===> Cluster delete configuration: %s", json)
+		log.Printf("===> cluster delete configuration: %s", json)
 	}
 
 	if _, err := meta.(*Client).ocean.Spark().DeleteCluster(ctx, input); err != nil {
 		return fmt.Errorf("[ERROR] onDelete() -> Failed to delete cluster: %s", err)
 	}
 
+	if forceDelete {
+		log.Printf("===> cluster has been force deleted: %s <===", resourceData.Id())
+		return nil
+	}
+
+	if err := waitUntilClusterDeleted(ctx, resourceData, meta); err != nil {
+		return fmt.Errorf("[ERROR] onDelete() -> failed to complete delete cluster: %s", err)
```

Comment:
```suggestion
		return err
```
here we are returning an error, not an error log string.

This gets passed up to the `return diag.FromErr(err)` call. We should leave the formatting of the error string to that mehod